releases/ReleaseCompatibility.md

```diff
@@ -0,0 +1,29 @@
+# Releases & Compatibility
+
+To date, we have been somewhat cavalier about making breaking changes in Fairlearn.
+Although we are currently pre-`v1.0.0` and hence without particular commitments to compatibility (as generally understood - see e.g. the [Semantic Versioning Scheme](https://semver.org/)), we should work to reduce the number of breaking changes we make.
+
+## The Problem
+
+At the time of writing, we are about to release `v0.4.6`, which has some substantial breaking changes in the Metrics area of Fairlearn.
+Before it, `v0.4.5` reworked a smaller subset of this functionality, which caused smaller breakages.
+There are also further changes to Metrics planned, which may well cause further breaks (although these should be more minor).
+All this is obviously undesirable from a user standpoint - these look like 'patch' level releases, but are actually breaking their code.
+Concretely, we've had users install Fairlearn with `pip`, and then find themselves unable to run Notebooks from our GitHub project - not because the functionality was missing, but because it had been renamed.
+
+We do not need deprecation policies as elaborate of those of [SciKit-Learn](https://numpy.org/neps/nep-0023-backwards-compatibility.html) or [NumPy](https://numpy.org/neps/nep-0023-backwards-compatibility.html) - indeed, policies such as those would be ridiculous given the size of our code and user bases.
+However, we do need to start to move in that direction, or we will not be able to grow our user base due to chaos in the code.
+
+## Proposed Solution
+
+Starting with `v0.5.0` we should make a commitment that anything which works at `v0.n.m_0` will also work for `v0.n.m` so long as `m >= m_0`.
+
+We will enforce this using builds which run the notebooks currently in the repository against versions of Fairlearn published on PyPI.
+We already have a build which checks that our notebooks can run against the latest published version of Fairlearn (due to the metrics changes, this build is a steady red); this would be an extension of that functionality.
+For full compliance with the above, we will have to devise a means of allowing each notebook to specify its `m_0`.
```

Comment:
I have started creating informally formatted "headers" in notebooks. See: [UCI Credit-card Default notebook](https://github.com/fairlearn/fairlearn/blob/master/notebooks/Binary%20Classification%20with%20the%20UCI%20Credit-card%20Default%20Dataset.ipynb). I wonder if we can just include that information there and then extract it in test/notebooks?

Reply:
We have that test/notebook directory with specific test cases per notebook anyway, so it seems easier to put that info there, unless you want users to see the versions that a notebook is compatible with (?) I could see some value in that as well.

Reply:
I'd tend towards putting the information in the notebook, and providing some Fairlearn method for doing a version check.

Reply:
Removed bit about using notebooks for testing. Just go for the tests instead.